### PR TITLE
Adding rate limitations for emitting data messages (can be set from D…

### PR DESCRIPTION
**Pull request by ElBabbler**
Resolving [this](https://github.com/elasticio/sailor-nodejs/issues/76) issue:
- Adding rate limitations for emitting data messages (can be set from DATA_RATE_LIMIT envar - 200/sec by default), error messages (can be set from ERROR_RATE_LIMIT envar - 20/sec by default) and snapshot messages (can be set from SNAPSHOT_RATE_LIMIT envar - 20/sec by default).

### CODE REVIEW

**Comment by zubairov** [comment]
@ElBabbler that was quick 👍 but tests are failing, what is the reason?

**Comment by ElBabbler** [comment]
Thank you, saw it. Now pushing correct version of that

**Comment by zubairov** [comment]
Please also change the version in the `package.json` and use the number after `.dev`, for example as it was done here:

https://github.com/elasticio/sailor-nodejs/commit/138dd8e1a720e053e1f8cd42e5228779a5e8b942#diff-b9cfc7f2cdf78a7f4b91a753d10865a2

File: package.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "elasticio-sailor-nodejs",
   "description": "The official elastic.io library for bootstrapping and executing for Node.js connectors",
-  "version": "2.2.2",
+  "version": "2.2.3.dev",
```
Comment (zubairov) [inline package.json]: should be `dev.1` not `dev`
```suggestion
  "version": "2.2.3.dev.1",
```

File: lib/amqp.js [hunk 2]
```diff
@@ -3,13 +3,25 @@ const amqplib = require('amqplib');
...
 const co = require('co');
 const _ = require('lodash');
+const rateLimit = require('function-rate-limit');
 
 const HEADER_ROUTING_KEY = 'x-eio-routing-key';
 const HEADER_ERROR_RESPONSE = 'x-eio-error-response';
 
 class Amqp {
     constructor(settings) {
         this.settings = settings;
```
Comment (zubairov) [inline lib/amqp.js]: where these settings are coming from? Environment variables?

**Comment by kirill-levitskiy-elasticio** [comment]
> Question on how to configure it. Environment variables?

Correct.
* DATA_RATE_LIMIT
* ERROR_RATE_LIMIT
* SNAPSHOT_RATE_LIMIT

**Comment by zubairov** [comment]
README update + tests are missing

**Comment by zubairov** [comment]
I just realised the implementation of this is completely wrong. Need to be rewritten using a promise based library https://www.npmjs.com/package/p-throttle

**Pull request merged**